Feature/add jira integration

merkata opened the pull request:
This feature adds an action to sync issues from Github to Jira. A JIRA_URL secrets is already created and the Automation on the side of Jira is configured as well. The sync is unidirectional, from Github to Jira, meaning creating/closing/reopening an issue on Github does that action on Jira, there is no Jira to Github direction.

arturo-seijas reviewed (approved):
LGTM

gregory-schiano commented:
Can't we ask beliaev-maksim to move the project under canonical namespace to ensure future maintenance ?

merkata commented:
> Can't we ask beliaev-maksim to move the project under canonical namespace to ensure future maintenance ?

That's a good point, I'll ask him. As we are evaluating, I propose to merge the PR as is and explore further.

jdkandersson reviewed (approved):
This will be interesting, looks like the CI/CD is failing though - need to fix that before merging

merkata commented:
> This will be interesting, looks like the CI/CD is failing though - need to fix that before merging

Thanks for spotting - the CI/CD is fixed, I cherry picked some commits from the Chaos PR, as I enabled Jira integration there, also trivy config for trivy scan to pass, then addressed linting issues - I created different PRs to have smaller codebase surface, though ultimately when they get merged, we are all green on CI/CD part.

jdkandersson commented:
I'm confused, it still looks like the status checks were failing
![Screenshot 2023-02-01 at 11 09 15 am](https://user-images.githubusercontent.com/51036209/215912899-0d4a5a6d-7e2c-4602-ace3-883ee2a1fc1c.jpg)

merkata commented:
> I'm confused, it still looks like the status checks were failing ![Screenshot 2023-02-01 at 11 09 15 am](https://user-images.githubusercontent.com/51036209/215912899-0d4a5a6d-7e2c-4602-ace3-883ee2a1fc1c.jpg)

I cherry picked commits from the Chaos test PR to split it, because in it I fixed linting issues, added a trivy config for the CI/CD to get to Chaos tests to verify they work, added Jira integration and then the whole PR was just too big and out of scope (the scope was just adjusting Chaos test variables). All the linting issues are fixed as part of the feat/linter PR and once aproved, they will land on `main`.
